lib/chromedriver.js

```diff
@@ -525,6 +530,32 @@ class Chromedriver extends events.EventEmitter {
     this.capabilities = toW3cCapNames(this.capabilities);
   }
 
+  /**
+   * Sync the protocol by reading the given output
+   *
+   * @param {string} out The output of WebDriver process start
+   */
+  detectWebDriverProtocol(out) {
+    if (this.driverVersion) {
+      // Nothing does if the protocol was already detected
+      return;
+    }
+
+    // also print chromedriver version to logs
+    // will output something like
+    //  Starting ChromeDriver 2.33.506106 (8a06c39c4582fbfbab6966dbb1c38a9173bfb1a2) on port 9515
+    // Or MSEdge:
+    //  Starting Microsoft Edge WebDriver 111.0.1661.41 (57be51b50d1be232a9e8186a10017d9e06b1fd16) on port 9515
+    const match = WEBDRIVER_VERSION_PATTERN.exec(out);
+    if (match && match.length === 3) {
+      this.log.debug(`${match[1]} version: '${match[2]}'`);
+      this.driverVersion = match[2];
+      this.syncProtocol(match[2]);
```

Comment:
not sure if we need additionally pass `match[2]` into this.syncProtocol because we already have it stored as this.driverVersion...

Perhaps it would make sense to make it more straightforward: 

```
this.driverVersion = this.syncProtocol(match[2])
```

and make this.syncProtocol to return the same version number that was passed to it

Reply:
Just did not want to handle this.driverVersion was null case in `syncProtocol`

Reply:
there might be an issue there where syncProtocol throws an exception, but driverVersion is already set. Then calling this method again won't work

Reply:
Added try/catch for safe